server/src/app/chat.service.ts

```diff
@@ -1,44 +1,23 @@
 import { Injectable } from '@nestjs/common';
-import {
-  ChatSession,
-  GenerativeModel,
-  GoogleGenerativeAI,
-} from '@google/generative-ai';
 
-import { ChatContent } from 'data-model';
+import  ollama from 'ollama';
+import { Message } from 'ollama';
+
 
 @Injectable()
 export class ChatService {
-  model: GenerativeModel;
-  chatSession: ChatSession;
-  constructor() {
-    const genAI = new GoogleGenerativeAI(process.env.API_KEY);
-    this.model = genAI.getGenerativeModel({ model: 'gemini-pro' });
-    this.chatSession = this.model.startChat({
-      history: [
-        {
-          role: 'user',
-          parts: `You're a poet. Respond to all questions with a rhyming poem.
-            What is the capital of California?
-          `,
-        },
-        {
-          role: 'model',
-          parts:
-            'If the capital of California is what you seek, Sacramento is where you ought to peek.',
-        },
-      ],
+
+  constructor() {}
+
+  async chat(chatMessages: Message[], chatClient: any = ollama): Promise<Message> {
```

Comment:
```suggestion
  async chat(chatMessages: Message[], chatClient = ollama): Promise<Message> {
```
sinon le type de `response` (ligne 14) est `any` et donc le retour de la fonction sera aussi `any`